cockatrice/src/window_sets.cpp

```diff
@@ -147,32 +147,18 @@ void WndSets::selectRow(int row)
     view->scrollTo(idx, QAbstractItemView::EnsureVisible);
 }
 
-void WndSets::actEnable()
+void WndSets::actToggle()
 {
     QModelIndexList rows = view->selectionModel()->selectedRows();
-    if(rows.empty())
-        return;
 
-    model->toggleRow(rows.first().row(), true);
-}
-
-void WndSets::actDisable()
-{
-    QModelIndexList rows = view->selectionModel()->selectedRows();
-    if(rows.empty())
-        return;
-
-    model->toggleRow(rows.first().row(), false);
-}
-
-void WndSets::actEnableAll()
-{
-    model->toggleAll(true);
+    foreach(QModelIndex i, rows)
+        model->toggleRow(i.row(), !model->getToggleStatus(i.row()));
```

Comment:
You should be able to add a helper function such that `model->toggleRow(i.row())` is correct.